go/vt/schemadiff/partitioning_analysis.go

```diff
@@ -0,0 +1,731 @@
+		"create table %s (id int) PARTITION BY RANGE (yearweek(%s, 3))",
+		"create table %s (id int) PARTITION BY RANGE (yearweek(%s, 4))",
+		"create table %s (id int) PARTITION BY RANGE (yearweek(%s, 5))",
+		"create table %s (id int) PARTITION BY RANGE (yearweek(%s, 6))",
+		"create table %s (id int) PARTITION BY RANGE (yearweek(%s, 7))",
+		"create table %s (id int) PARTITION BY RANGE (year(%s))",
+	}
+	if is84 {
+		supportedVariations = append(supportedVariations, "create table %s (id int) PARTITION BY RANGE (unix_timestamp(%s))")
+	}
+	supportedVariationsEntities := []*CreateTableEntity{}
+	for _, supportedVariation := range supportedVariations {
+		query := fmt.Sprintf(supportedVariation,
+			sqlparser.CanonicalString(createTableEntity.CreateTable.GetTable().Name),
+			sqlparser.CanonicalString(colName),
+		)
+		cte, err := NewCreateTableEntityFromSQL(createTableEntity.Env, query)
+		if err != nil {
+			return false, false, err
+		}
+		supportedVariationsEntities = append(supportedVariationsEntities, cte)
+	}
+	for i, cte := range supportedVariationsEntities {
+		if sqlparser.Equals.Expr(expr, cte.CreateTable.TableSpec.PartitionOption.Expr) {
+			return true, i > 0, nil
+		}
+	}
+	return false, false, nil
+}
+
+// extractFuncWeekMode extracts the mode argument from a WEEK/YEARWEEK function, if applicable.
+// It returns a ModeUndefined when not applicable.
+func extractFuncWeekMode(funcExpr *sqlparser.FuncExpr) (int, error) {
+	switch funcExpr.Name.Lowered() {
+	case "week", "yearweek":
+	default:
+		return WeekModeUndefined, nil
+	}
+	if len(funcExpr.Exprs) <= 1 {
+		return 0, nil
+	}
+	// There is a `mode` argument in the YEARWEEK function.
+	literal, ok := funcExpr.Exprs[1].(*sqlparser.Literal)
+	if !ok {
+		return 0, fmt.Errorf("expected literal value in %v function", sqlparser.CanonicalString(funcExpr))
+	}
+	if literal.Type != sqlparser.IntVal {
+		return 0, fmt.Errorf("expected integer literal argument in %v function", sqlparser.CanonicalString(funcExpr))
+	}
+	intval, err := strconv.ParseInt(literal.Val, 0, 64)
+	if err != nil {
+		return 0, err
+	}
+	return int(intval), nil
+}
+
+// AnalyzeTemporalRangePartitioning analyzes a table for temporal range partitioning.
+func AnalyzeTemporalRangePartitioning(createTableEntity *CreateTableEntity) (*TemporalRangePartitioningAnalysis, error) {
+	analysis := &TemporalRangePartitioningAnalysis{}
+	withReason := func(msg string, args ...any) (*TemporalRangePartitioningAnalysis, error) {
+		analysis.Reason = fmt.Sprintf(msg, args...)
+		return analysis, nil
+	}
+	getColumn := func(lowered string) *ColumnDefinitionEntity {
+		// The column will exist, because we invoke Validate() before this.
+		return createTableEntity.ColumnDefinitionEntitiesMap()[lowered]
+	}
+	if err := createTableEntity.validate(); err != nil {
+		return nil, err
+	}
+	if !IsRangePartitioned(createTableEntity.CreateTable) {
+		return withReason("Table does not use PARTITION BY RANGE")
```

Comment:
Nit, but it looks like these strings end up becoming errors and errors aren't supposed to use sentence capitalization or punctuation.

Reply:
fixed.